IPS-279 Dynatrace Integration into Core-Front.

vam-gds opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed
IPS-279 Dynatrace integration into core-front.

Updated Code owners
Fixed dynatrace layer Dockerfile
core-front gha updated to log into dynatrace registry

Added dev-deploy directory, this contains the original Dockerfile without the dynatrace integration specifically for dev deployments

### Why did it change

IPS-279 Dynatrace integration into core-front.

Updated Code owners
Fixed dynatrace layer Dockerfile
core-front gha updated to log into dynatrace registry

Added dev-deploy directory, this contains the original Dockerfile without the dynatrace integration specifically for dev deployments

            modified:   .github/workflows/secure-post-merge.yml
            modified:   CODEOWNERS
            modified:   Dockerfile
            modified:   deploy/template.yaml
            new file:   dev-deploy/Dockerfile
### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [IPS-279](https://govukverify.atlassian.net/browse/IPS-279)

## Checklists

### Environment variables or secrets

<!-- Delete if changes DO include new environment variables or secrets -->
- [ ] No environment variables or secrets were added or changed

<!-- Delete if changes DO NOT include new environment variables or secrets -->
- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks


[IPS-279]: https://govukverify.atlassian.net/browse/IPS-279?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

two-first-names commented on Dockerfile:
```diff
@@ -29,6 +29,10 @@ COPY --chown=appuser:appgroup --from=builder /app/src ./src
 COPY --chown=appuser:appgroup --from=builder /app/yarn.lock ./
 
+# Add in dynatrace layer
+COPY --from=khw46367.live.dynatrace.com/linux/oneagent-codemodules:<technology> / /
```
```suggestion
COPY --from=khw46367.live.dynatrace.com/linux/oneagent-codemodules-musl:nodejs / /
```

vam-gds replied:
should be fixed now.

vam-gds committed:
IPS-279 Dynatrace integration into core-front.

Updated Code owners
Fixed dynatrace layer Dockerfile
core-front gha updated to log into dynatrace registry

Added dev-deploy directory
This contains the original Dockerfile without the
dynatrace integration specifically for dev deployments.

Added a step to log into dynatrace container reg

	modified:   .github/workflows/pre-merge-checks.yml
	modified:   .github/workflows/secure-post-merge.yml
	modified:   CODEOWNERS
	modified:   Dockerfile
	modified:   deploy/template.yaml
	new file:   dev-deploy/Dockerfile

vam-gds commented on deploy/template.yaml:
```diff
@@ -616,7 +622,7 @@ Resources:
               Value: !If
                 - IsBuild
                 - "development"
-                - !If [IsDevelopment, !Sub "development", !Ref "AWS::NoValue"]
```
please note this line failed in cfn-lint was something else meant to be added to be substituted if the condition IsDevelopment would evaluate to true ? Or was this just an oversight? I've fixed it, it it was just an oversight.
